pkgs/os-specific/linux/conky/default.nix

```
@@ -166,46 +159,49 @@ stdenv.mkDerivation rec {
     ++ lib.optional wirelessSupport wirelesstools
     ++ lib.optional curlSupport curl
     ++ lib.optional rssSupport libxml2
-    ++ lib.optional weatherXoapSupport libxml2
     ++ lib.optional nvidiaSupport libXNVCtrl
     ++ lib.optional pulseSupport libpulseaudio
     ++ lib.optional journalSupport systemd;
 
-  cmakeFlags =
-    [ ]
-    ++ lib.optional docsSupport "-DMAINTAINER_MODE=ON"
-    ++ lib.optional curlSupport "-DBUILD_CURL=ON"
-    ++ lib.optional (!ibmSupport) "-DBUILD_IBM=OFF"
-    ++ lib.optional imlib2Support "-DBUILD_IMLIB2=ON"
-    ++ lib.optional luaCairoSupport "-DBUILD_LUA_CAIRO=ON"
-    ++ lib.optional luaImlib2Support "-DBUILD_LUA_IMLIB2=ON"
-    ++ lib.optional (!mpdSupport) "-DBUILD_MPD=OFF"
-    ++ lib.optional (!ncursesSupport) "-DBUILD_NCURSES=OFF"
-    ++ lib.optional rssSupport "-DBUILD_RSS=ON"
-    ++ lib.optional (!x11Support) "-DBUILD_X11=OFF"
-    ++ lib.optional waylandSupport "-DBUILD_WAYLAND=ON"
-    ++ lib.optional xdamageSupport "-DBUILD_XDAMAGE=ON"
-    ++ lib.optional doubleBufferSupport "-DBUILD_XDBE=ON"
-    ++ lib.optional weatherMetarSupport "-DBUILD_WEATHER_METAR=ON"
-    ++ lib.optional weatherXoapSupport "-DBUILD_WEATHER_XOAP=ON"
-    ++ lib.optional wirelessSupport "-DBUILD_WLAN=ON"
-    ++ lib.optional nvidiaSupport "-DBUILD_NVIDIA=ON"
-    ++ lib.optional pulseSupport "-DBUILD_PULSEAUDIO=ON"
-    ++ lib.optional journalSupport "-DBUILD_JOURNAL=ON";
-
-  # `make -f src/CMakeFiles/conky.dir/build.make src/CMakeFiles/conky.dir/conky.cc.o`:
-  # src/conky.cc:137:23: fatal error: defconfig.h: No such file or directory
-  enableParallelBuilding = false;
-
-  doCheck = true;
-
-  meta = with lib; {
+  cmakeFlags = [
+    (lib.cmakeBool "REPRODUCIBLE_BUILD" true)
+    (lib.cmakeBool "RELEASE" true)
+    (lib.cmakeBool "BUILD_TESTING" finalAttrs.doCheck)
```

```suggestion
    (lib.cmakeBool "BUILD_TESTING" finalAttrs.finalPackage.doCheck)
```

i believe `finalAttrs.finalPackage.doCheck` is actual, computed `doCheck` and using that here should let you remove the explicit `doCheck = stdenv.buildPlatform.canExecute ...` further down.

---

Hm, interesting. I do have do still set `doCheck = true` or `BUILD_TESTING` is still false, is that expected?

---

oh, i guess you do still have to set `doCheck = true` since [the default](https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/make-derivation.nix#L291) is false :(

---

Still nicer than having to recreate the cross check myself :)